railties/lib/rails/commands/credentials/credentials_command.rb

```diff
@@ -91,7 +92,7 @@ def ensure_credentials_have_been_added
 
         def change_credentials_in_system_editor
           credentials.change do |tmp_path|
-            system("#{ENV["EDITOR"]} #{tmp_path}")
+            system("#{ENV["EDITOR"]} #{Shellwords.escape(tmp_path)}")
```

Comment:
Have you considered `system(ENV["EDITOR"], tmp_path)`, that would perfectly take care of the escaping and avoid a subshell.

Reply:
I did, but then this test failed

https://github.com/rails/rails/blob/d364cfb34ea5d57f314e6e1449a8ca63029623a7/railties/test/commands/credentials_test.rb#L41

https://buildkite.com/rails/rails/builds/79105#5b1fa71b-14c0-4480-a298-afe249b6f6e2

Reply:
Ah!